cachelib/allocator/memory/SlabAllocator.h

```diff
@@ -246,19 +247,20 @@ class SlabAllocator {
         static_cast<uint32_t>(reinterpret_cast<const uint8_t*>(ptr) -
                               reinterpret_cast<const uint8_t*>(slab)) /
         allocSize;
-    return CompressedPtr{slabIndex, allocIdx};
+    return CompressedPtr{slabIndex, allocIdx, isMultiTiered};
```

Comment:
So, tier id is not set here. How are you going to provide the tier id?

I think you are going to have a separate `MemoryAllocator` per tier, so separate `SlabAllocator`. If so, I think the tier ID could be a member variable of the `SlabAllocator`? 

What about the `PtrCompressor` then? Would `CacheAllocator` create separate `PtrCompressor` per `MemoryAllocator`?

Reply:
Please refer to this commit https://github.com/intel/CacheLib/pull/56/commits/2704ac894d2a4fcf7be29d38784abf2e0fbc6396#diff-a6542b6dbf2cfb5e03e82205ee960757ab2b50de7bc25085089a3cffba40ae87 which shows how tier Id is passed to the alloctor.compress() methods.

This change is due to be sent for upstream review soon.